src/Functions/h3PolygonToCells.cpp

```diff
@@ -0,0 +1,187 @@
+#include "config.h"
+
+#if USE_H3
+
+#include <Columns/ColumnArray.h>
+
+#include <Functions/FunctionFactory.h>
+#include <Functions/geometryConverters.h>
+#include <Functions/IFunction.h>
+
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/point_xy.hpp>
+
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypesNumber.h>
+
+#include <constants.h>
+#include <h3api.h>
+
+
+// TODO: from h3ToChildren, why this exact number?
+static constexpr size_t MAX_ARRAY_SIZE = 1 << 30;
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ARGUMENT_OUT_OF_BOUND;
+    extern const int TOO_LARGE_ARRAY_SIZE;
+    extern const int ILLEGAL_COLUMN;
+}
+
+class FunctionH3PolygonToCells : public IFunction
+{
+public:
+    static constexpr auto name = "h3PolygonToCells";
+    String getName() const override { return name; }
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionH3PolygonToCells>(); }
+
+    size_t getNumberOfArguments() const override { return 2; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+    // TODO: Is it suitable?
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes &) const override
+    {
+        return std::make_shared<DataTypeArray>(std::make_shared<DataTypeUInt64>());
+    }
+
+    ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
+    {
+        // TODO: what is non-const in this context?
+        //       and why are h3 functions using non-const
```

Comment:
Non-const are the parameters which come from columns, not from the constant provided in SQL. For example `h3PolygonToCells(x)` where x is a column from a table vs. `h3PolygonToCells([1.23, 4.56])`.